src/coap_openssl.c

```diff
@@ -835,7 +837,7 @@ void coap_dtls_handle_timeout(coap_session_t *session) {
   SSL *ssl = (SSL *)session->tls;
 
   assert(ssl != NULL);
-  if (DTLSv1_handle_timeout(ssl) < 0) {
+  if ((session->state == COAP_SESSION_STATE_HANDSHAKE && ++session->dtls_timeout_count > session->max_retransmit) || DTLSv1_handle_timeout(ssl) < 0) {
```

Comment:
Can you please enclose the relations in parentheses and add a line break before `||` for clarity and improved readbility?